Adds running an RPC using systemd article (#104)
moondance-labs/tanssi-docs: node-operators/appchain-node/rpc-systemd.md

---
title: Run an RPC Node Using Systemd
description: Learn how to set up and run a Tanssi Appchain node using Systemd, which allows you to have your own RPC endpoint to interact with your Appchain.
---

# Run an Appchain Node Using Systemd

## Introduction {: #introduction }

Running a Tanssi Appchain node allows you to connect to and interact with the network using your infrastructure via either HTTP or WebSocket protocols. 

Nodes store block data and network state. However, developers can run different kinds of nodes:
 

>  - **Full Archive Node** - a node storing the entire block data and network state at all block heights. Such nodes are helpful when querying historical data from old blocks. However, a full archive node takes up a lot of space

Reviewer: This should be a text snippet that is recycled between Docker and SystemD

- **Appchain specifications file** - the Appchain specification file is needed to run the node. You can download it from the dashboard in the [dApp](https://apps.tanssi.network/){target=\_blank} by clicking the `Appchain Data` link

    ![Getting the chain specs](/images/node-operators/appchain-node/rpc-systemd/rpc-systemd-1.webp)

- **Relay chain specifications file** - the relay chain specification file can be downloaded from this [public GitHub repository](https://github.com/papermoonio/external-files/blob/main/Moonbeam/Moonbase-Alpha/){target=\_blank}

## Download the Latest Release {: #download-latest-release }

Every new release includes two different node binaries, one for EVM-compatible Appchains and another one for Substrate Appchains. To get started, run the following command to get the latest release binary that matches your Appchain type and make it executable:

=== "EVM-Compatible Appchain"

    ```bash
    wget https://github.com/moondance-labs/tanssi/releases/latest/download/container-chain-template-frontier-node && \

>     chmod +x ./container-chain-template-frontier-node

Reviewer: I think it is important that we state somewhere that these steps must be adapted to your specific infra-running configuration or something like that.

For example, in Moonbeam we do it via a `moonbeam.service` and then we do `chmod` but a bit differently (for the service only, not general. 

https://docs.moonbeam.network/node-operators/networks/run-a-node/systemd/#setup-the-service

Author: Added a few lines in the intro, inviting the reader to adapt the guide

    ```

=== "Substrate Appchain"

    ```bash
    wget https://github.com/moondance-labs/tanssi/releases/latest/download/container-chain-template-simple-node && \
    chmod +x ./container-chain-template-simple-node
    ```

!!! note

>     Optimized binary versions for [Skylake](https://www.intel.com/content/www/us/en/products/platforms/details/skylake-u-y.html){target=\_blank} and [Zen3](https://www.amd.com/en/technologies/zen-core){target=\_blank} architectures are also available in the [releases](https://github.com/moondance-labs/tanssi/releases/latest){target=\_blank} page.

Reviewer: ```suggestion
    It is recommended that you run the optimized binary versions for either [Skylake](https://www.intel.com/content/www/us/en/products/platforms/details/skylake-u-y.html){target=\_blank} or [Zen3](https://www.amd.com/en/technologies/zen-core){target=\_blank} architectures. They are available in the [releases](https://github.com/moondance-labs/tanssi/releases/latest){target=\_blank} page.
```

## Download the Relay Chain Specs File {: #download-relay-specs }

The node binary file includes also the necessary code to run a relay chain node. When launching your Appchain's node, it will also be required to provide the relay chain's specification file as a parameter. 

Download the relay chain specification file executing:

```bash
wget https://raw.githubusercontent.com/papermoonio/external-files/main/Moonbeam/Moonbase-Alpha/westend-alphanet-raw-specs.json
```

## Setup the Systemd Service {: #setup-systemd-service }

[Systemd](https://systemd.io/){target=\_blank} is a management system for Linux systems that manages services (daemons in Unix-like systems jargon), starting them automatically when the computer starts or reboots, or restarting them upon unexpected failures.

> It is a good practice to have the service running with its own non-root account and grant that account writing access to a specific directory. Run the following commands to configure a new account and the directory:

Reviewer: I think this should be stated earlier tbh

Author: [Moved to the intro]

```bash
sudo chown -R appchain_node_service /var/lib/appchain-data
```

And finally, move the binary and the relay chain spec to the folder:

```bash
mv ./container-chain-template-*-node /var/lib/appchain-data && \
mv ./westend-alphanet-raw-specs.json /var/lib/appchain-data
```

!!! note
    To keep all the necessary files grouped in the same directory, it is recommended to also copy there your Appchain's specification file.

### Create the Systemd Service configuration file {: #create-systemd-configuration }

The next step is to create the Systemd configuration file. 

You can create the file by running the following command:

```bash
sudo touch /etc/systemd/system/appchain.service
```

Now you can open the file using your favorite text editor (vim, emacs, nano, etc) and add the configuration for the service:

```bash
[Unit]
Description="Appchain systemd service"
After=network.target
StartLimitIntervalSec=0

[Service]
Type=simple
Restart=on-failure
RestartSec=10
User=appchain_node_service
SyslogIdentifier=appchain
SyslogFacility=local7
KillSignal=SIGHUP
ExecStart=/var/lib/appchain-data/container-chain-template-APPCHAIN_TYPE-node \

> --chain=YOUR_APPCHAIN_SPECS_FILE_LOCATION \

Reviewer: Use the same structure as with [Docker](https://docs.tanssi.network/node-operators/appchain-node/rpc-docker/#start-up-command).

That is:
- Explain Data they have to modify and then show the content of the file. Also, put the content of the flags in the same order as Docker

--bootnodes=/dns4/frag3-stagenet-relay-val-2.g.moondev.network/tcp/30334/p2p/12D3KooWMAtGe8cnVrg3qGmiwNjNaeVrpWaCTj82PGWN7PBx2tth \
--bootnodes=/dns4/frag3-stagenet-relay-val-3.g.moondev.network/tcp/30334/p2p/12D3KooWLKAf36uqBBug5W5KJhsSnn9JHFCcw8ykMkhQvW7Eus3U \
--bootnodes=/dns4/vira-stagenet-relay-validator-0.a.moondev.network/tcp/30334/p2p/12D3KooWSVTKUkkD4KBBAQ1QjAALeZdM3R2Kc2w5eFtVxbYZEGKd \
--bootnodes=/dns4/vira-stagenet-relay-validator-1.a.moondev.network/tcp/30334/p2p/12D3KooWFJoVyvLNpTV97SFqs91HaeoVqfFgRNYtUYJoYVbBweW4 \
--bootnodes=/dns4/vira-stagenet-relay-validator-2.a.moondev.network/tcp/30334/p2p/12D3KooWP1FA3dq1iBmEBYdQKAe4JNuzvEcgcebxBYMLKpTNirCR \
--bootnodes=/dns4/vira-stagenet-relay-validator-3.a.moondev.network/tcp/30334/p2p/12D3KooWDaTC6H6W1F4NkbaqK3Ema3jzc2BbhE2tyD3YEf84yNLE 

[Install]
WantedBy=multi-user.target
```

Note that the `ExecStart` command still needs some parameters to be changed before starting the service. The following section focuses on this command and its optional flags.

## The `ExecStart` Command {: #execstart-command }

The previous `ExecStart` example has some parameters that need to be changed to match your specific Appchain:

> - `EVM compatibility` - Tanssi releases two different binaries, one for EVM-compatible Appchains and another for only Substrate Appchains. Replace `APPCHAIN_TYPE` with either `frontier` for EVM Appchains or `simple` for Substrate Appchains

Reviewer: Goes before, similar to the Docker structure. Use the exact same format so that it looks cohesive

- `Specification file` - replace `YOUR_APPCHAIN_SPECS_FILE_LOCATION` with your Appchain's absolute path. If the file was copied in the same directory as the binary file and the relay chain specs, then your path will look like `/var/lib/appchain-data/YOUR_FILENAME.json`, e.g. `/var/lib/appchain-data/spec-raw.json`
- `Bootnode` - a bootnode is a full archive node that is used to sync the network from scratch. You'll need to [retrieve your Tanssi Appchain bootnode](#fetching-bootnode-information) and replace `INSERT_YOUR_APPCHAIN_BOOTNODE` with the actual bootnode information

> ### Fetching Bootnode Information {: #fetching-bootnode-information}

Reviewer: This should be a text snippet and recycled between Docker and Systemd

Bootnode information can be read from the Tanssi Appchain storage on its [Polkadot.js explorer](https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Ffraa-dancebox-rpc.a.dancebox.tanssi.network#/chainstate){target=\_blank}.

To do so, take the following steps:

1. Select `dataPreservers` as the module to query
2. Set the storage query to `bootNodes`
3. Provide your Tanssi Appchain ID
4. Click on the **+** sign

![Getting the bootnode](/images/node-operators/appchain-node/rpc-systemd/rpc-systemd-2.webp)

> ### Run Flags {: #run-flags }

Reviewer: Text snippet that is recycled between Docker and SystemD. Maybe not the intro as you specifically state `execStart` but the other part yes

    ```bash
    /var/lib/appchain-data/container-chain-template-frontier-node --help
    ```

=== "Simple Substrate Appchain"

    ```bash
    /var/lib/appchain-data/container-chain-template-simple-node --help
    ```

### Full Node Configuration Example for the Demo EVM Appchain {: #example-demo-evm-appchain}

The following example is a fully functional full-node configuration for the [demo EVM Appchain](/builders/tanssi-network/networks/dancebox/demo-evm-containerchain/){target=\_blank} deployed on Dancebox with an ID of `3001`. 

The raw chain specification file for the demo Appchain is required to run the node, and can be downloaded from this [public GitHub repository](https://github.com/papermoonio/external-files/blob/main/Tanssi/Demo-EVM-Appchain/){target=\_blank}. Download the file and place it in the `/var/lib/appchain-data/` directory.

```bash
[Unit]
Description="Appchain systemd service"
After=network.target
StartLimitIntervalSec=0

[Service]
Type=simple
Restart=on-failure
RestartSec=10
User=appchain_node_service
SyslogIdentifier=appchain
SyslogFacility=local7
KillSignal=SIGHUP
ExecStart=/var/lib/appchain-data/container-chain-template-frontier-node \

> --chain=/var/lib/appchain-data/container-3001-raw-specs.json \

Reviewer: similar comment as before: use the same structure and order of flags for docker. In fact, I just realized that the Demo EVM example is not following the same structure than the `Start-Up Command` section, can you fix that plz?

--bootnodes=/dns4/frag3-stagenet-relay-val-3.g.moondev.network/tcp/30334/p2p/12D3KooWLKAf36uqBBug5W5KJhsSnn9JHFCcw8ykMkhQvW7Eus3U \
--bootnodes=/dns4/vira-stagenet-relay-validator-0.a.moondev.network/tcp/30334/p2p/12D3KooWSVTKUkkD4KBBAQ1QjAALeZdM3R2Kc2w5eFtVxbYZEGKd \
--bootnodes=/dns4/vira-stagenet-relay-validator-1.a.moondev.network/tcp/30334/p2p/12D3KooWFJoVyvLNpTV97SFqs91HaeoVqfFgRNYtUYJoYVbBweW4 \
--bootnodes=/dns4/vira-stagenet-relay-validator-2.a.moondev.network/tcp/30334/p2p/12D3KooWP1FA3dq1iBmEBYdQKAe4JNuzvEcgcebxBYMLKpTNirCR \
--bootnodes=/dns4/vira-stagenet-relay-validator-3.a.moondev.network/tcp/30334/p2p/12D3KooWDaTC6H6W1F4NkbaqK3Ema3jzc2BbhE2tyD3YEf84yNLE 

[Install]
WantedBy=multi-user.target
```

## Run the Service {: #run-the-service }

Finally, enable the service and start it for the first time:

```bash
systemctl enable appchain.service && \
systemctl start appchain.service
```

You can verify that the service is up and running correctly running:

```bash
systemctl status appchain.service
```

--8<-- 'code/node-operators/appchain-node/rpc-systemd/terminal/check-status.md'

And check the logs, if needed, with the following command:

```bash

> journalctl -f -u appchain.service

Reviewer: Add sample logs

---
title: Run an RPC Node Using Systemd
description: Learn how to set up and run a Tanssi Appchain node using Systemd, which allows you to have your own RPC endpoint to interact with your Appchain.
---

# Run an Appchain Node Using Systemd

## Introduction {: #introduction }

--8<-- 'text/node-operators/appchain-node/intro.md'

In this guide, you'll learn how to spin up a Tanssi Appchain node using a binary executable file and manage the service with [Systemd](https://systemd.io/){target=\_blank} on Linux systems.

> The article follows the good practice of running the service with its own non-root account and granting that account writing access to a specific directory. However, do adapt this article's steps and instructions to your infrastructure configuration, preferences, and security policies.

Reviewer 2: ```suggestion
The article follows the good practice of running the service with its own non-root account and granting that account write access to a specific directory. However, you can adapt this article's steps and instructions to your infrastructure configuration, preferences, and security policies.
```

!!! note
    It is not possible to run an RPC node for Snap Appchains as they run on a private network, and their nodes are, therefore, unreachable for syncing.

## Checking Prerequisites {: #checking-prerequisites }

To get started, you'll need access to a computer running an Ubuntu Linux OS and root privileges. You will also need:

- **Node binary file** - the instructions in this guide execute the [latest](https://github.com/moondance-labs/tanssi/releases/latest){target=\_blank} official stable node release. If you want to build and run your own file, make sure to follow the instructions for [building your Appchain node](/builders/build/customize/prerequisites){target=\_blank}

> - **Appchain specifications file** - the Appchain specification file is needed to run the node. You can download it from the dashboard in the [dApp](https://apps.tanssi.network/){target=\_blank} by clicking the `Appchain Data` link

Reviewer 2: ```suggestion
- **Appchain specifications file** - the Appchain specification file is needed to run the node. You can download it from the dashboard in the [dApp](https://apps.tanssi.network/){target=\_blank} by clicking the **Appchain Data** link
```

When referring to UI elements on a page, you can use bold formatting

>     ![Getting the chain specs](/images/node-operators/appchain-node/rpc-systemd/rpc-systemd-1.webp)

Reviewer 2: Just curious - do you know what zoom % you use when taking screenshots? I feel like these screenshots are a bit blurry and I'm not sure if it's like a webp thing or the size of the screenshot vs on the docs

Reviewer 2: I also prefer, if possible, having the browser in the screenshot instead of just part of the screen

Author: hmm, I wasn't using any zoom at all 
Maybe low res in the export settings? Now I replaced the image and exported the webp using the same setting (3300x1819 px, 300DPI). it looks good to me, but let me know otherwise!

Reviewer 2: looks great!! it probably doesn't help that i'm viewing it on a laptop screen

- **Relay chain specifications file** - the relay chain specification file can be downloaded from this [public GitHub repository](https://github.com/papermoonio/external-files/blob/main/Moonbeam/Moonbase-Alpha/){target=\_blank}

## Download the Latest Release {: #download-latest-release }

> Every new release includes two different node binaries, one for EVM-compatible Appchains and another one for Substrate Appchains. To get started, run the following command to get the latest release binary that matches your Appchain type and make it executable:

Reviewer 2: ```suggestion
Every new release includes two node binaries, one for EVM-compatible Appchains and another for Substrate Appchains. To get started, run the following command to get the latest release binary that matches your Appchain type and make it executable:
```

=== "EVM-Compatible Appchain"

    ```bash
    wget https://github.com/moondance-labs/tanssi/releases/latest/download/container-chain-template-frontier-node && \
    chmod +x ./container-chain-template-frontier-node
    ```

=== "Substrate Appchain"

    ```bash
    wget https://github.com/moondance-labs/tanssi/releases/latest/download/container-chain-template-simple-node && \
    chmod +x ./container-chain-template-simple-node
    ```

!!! note
    It is recommended that you run the optimized binary versions for either [Skylake](https://www.intel.com/content/www/us/en/products/platforms/details/skylake-u-y.html){target=\_blank} or [Zen3](https://www.amd.com/en/technologies/zen-core){target=\_blank} architectures. They are available in the [releases](https://github.com/moondance-labs/tanssi/releases/latest){target=\_blank} page.

## Download the Relay Chain Specs File {: #download-relay-specs }

The node binary file includes also the necessary code to run a relay chain node. When launching your Appchain's node, it will also be required to provide the relay chain's specification file as a parameter. 

> Download the relay chain specification file executing:

Reviewer 2: ```suggestion
Download the relay chain specification file by executing:
```

```bash
wget https://raw.githubusercontent.com/papermoonio/external-files/main/Moonbeam/Moonbase-Alpha/westend-alphanet-raw-specs.json
```

## Setup the Systemd Service {: #setup-systemd-service }

[Systemd](https://systemd.io/){target=\_blank} is a management system for Linux systems that manages services (daemons in Unix-like systems jargon), starting them automatically when the computer starts or reboots, or restarting them upon unexpected failures. The following commands configure a new account, the directory, and move the previously downloaded files to the right location.

Create a new account to run the service:

```bash
adduser appchain_node_service --system --no-create-home
```

Create a directory to store the required files and data:

```bash
mkdir /var/lib/appchain-data
```

Set the folder's ownership to the account that will run the service to ensure writing permission:

```bash
sudo chown -R appchain_node_service /var/lib/appchain-data
```

And finally, move the binary and the relay chain spec to the folder:

```bash
mv ./container-chain-template-*-node /var/lib/appchain-data && \
mv ./westend-alphanet-raw-specs.json /var/lib/appchain-data
```

!!! note

>     To keep all the necessary files grouped in the same directory, it is recommended to also copy there your Appchain's specification file.

Reviewer 2: ```suggestion
    To keep all the necessary files grouped in the same directory, it is also recommended to copy your Appchain's specification file there.
```

> ### Create the Systemd Service configuration file {: #create-systemd-configuration }

Reviewer 2: ```suggestion
### Create the Systemd Service Configuration File {: #create-systemd-configuration }
```

The next step is to create the Systemd configuration file. 

You can create the file by running the following command:

```bash
sudo touch /etc/systemd/system/appchain.service
```

> Now you can open the file using your favorite text editor (vim, emacs, nano, etc) and add the configuration for the service.

Reviewer 2: ```suggestion
Now, you can open the file using your favorite text editor (vim, emacs, nano, etc.) and add the configuration for the service.
```

Note that the `ExecStart` command  has some parameters that need to be changed to match your specific Appchain:

> - `Specification file` - replace `YOUR_APPCHAIN_SPECS_FILE_LOCATION` with your Appchain's absolute path. If the file was copied in the same directory as the binary file and the relay chain specs, then your path will look like `/var/lib/appchain-data/YOUR_FILENAME.json`, e.g. `/var/lib/appchain-data/spec-raw.json`

Reviewer 2: ```suggestion
- `Specification file` - replace `YOUR_APPCHAIN_SPECS_FILE_LOCATION` with your Appchain's absolute path. If you copied the file in the same directory as the binary file and the relay chain specs, then your path will look like `/var/lib/appchain-data/YOUR_FILENAME.json`, e.g., `/var/lib/appchain-data/spec-raw.json`
```

--bootnodes=/dns4/frag3-stagenet-relay-val-1.g.moondev.network/tcp/30334/p2p/12D3KooWQYLjopFtjojRBfTKkLFq2Untq9yG7gBjmAE8xcHFKbyq \
--bootnodes=/dns4/frag3-stagenet-relay-val-2.g.moondev.network/tcp/30334/p2p/12D3KooWMAtGe8cnVrg3qGmiwNjNaeVrpWaCTj82PGWN7PBx2tth \
--bootnodes=/dns4/frag3-stagenet-relay-val-3.g.moondev.network/tcp/30334/p2p/12D3KooWLKAf36uqBBug5W5KJhsSnn9JHFCcw8ykMkhQvW7Eus3U \
--bootnodes=/dns4/vira-stagenet-relay-validator-0.a.moondev.network/tcp/30334/p2p/12D3KooWSVTKUkkD4KBBAQ1QjAALeZdM3R2Kc2w5eFtVxbYZEGKd \
--bootnodes=/dns4/vira-stagenet-relay-validator-1.a.moondev.network/tcp/30334/p2p/12D3KooWFJoVyvLNpTV97SFqs91HaeoVqfFgRNYtUYJoYVbBweW4 \
--bootnodes=/dns4/vira-stagenet-relay-validator-2.a.moondev.network/tcp/30334/p2p/12D3KooWP1FA3dq1iBmEBYdQKAe4JNuzvEcgcebxBYMLKpTNirCR \
--bootnodes=/dns4/vira-stagenet-relay-validator-3.a.moondev.network/tcp/30334/p2p/12D3KooWDaTC6H6W1F4NkbaqK3Ema3jzc2BbhE2tyD3YEf84yNLE 

[Install]
WantedBy=multi-user.target
```

### Run Flags {: #run-flags }

The flags used in the `ExecStart` command can be adjusted according to your preferences and hardware configuration. The following ones are some of the most note-worthy:

> --8<-- 'code/node-operators/appchain-node/rpc-systemd/terminal/run-flags.md'

Reviewer 2: ```suggestion
--8<-- 'text/node-operators/appchain-node/run-flags.md'
```

    --bootnodes=/dns4/frag3-stagenet-relay-val-1.g.moondev.network/tcp/30334/p2p/12D3KooWQYLjopFtjojRBfTKkLFq2Untq9yG7gBjmAE8xcHFKbyq \
    --bootnodes=/dns4/frag3-stagenet-relay-val-2.g.moondev.network/tcp/30334/p2p/12D3KooWMAtGe8cnVrg3qGmiwNjNaeVrpWaCTj82PGWN7PBx2tth \
    --bootnodes=/dns4/frag3-stagenet-relay-val-3.g.moondev.network/tcp/30334/p2p/12D3KooWLKAf36uqBBug5W5KJhsSnn9JHFCcw8ykMkhQvW7Eus3U \
    --bootnodes=/dns4/vira-stagenet-relay-validator-0.a.moondev.network/tcp/30334/p2p/12D3KooWSVTKUkkD4KBBAQ1QjAALeZdM3R2Kc2w5eFtVxbYZEGKd \
    --bootnodes=/dns4/vira-stagenet-relay-validator-1.a.moondev.network/tcp/30334/p2p/12D3KooWFJoVyvLNpTV97SFqs91HaeoVqfFgRNYtUYJoYVbBweW4 \
    --bootnodes=/dns4/vira-stagenet-relay-validator-2.a.moondev.network/tcp/30334/p2p/12D3KooWP1FA3dq1iBmEBYdQKAe4JNuzvEcgcebxBYMLKpTNirCR \
    --bootnodes=/dns4/vira-stagenet-relay-validator-3.a.moondev.network/tcp/30334/p2p/12D3KooWDaTC6H6W1F4NkbaqK3Ema3jzc2BbhE2tyD3YEf84yNLE 

    [Install]
    WantedBy=multi-user.target
    ```

--8<-- 'text/node-operators/appchain-node/fetching-bootnode-section.md'

### Full Node Configuration Example for the Demo EVM Appchain {: #example-demo-evm-appchain}

> The following example is a fully functional full-node configuration for the [demo EVM Appchain](/builders/tanssi-network/networks/dancebox/demo-evm-containerchain/){target=\_blank} deployed on Dancebox with an ID of `3001`. 

Reviewer: ```suggestion
The following example deploys a fully functional full archive node for the [demo EVM Appchain](/builders/tanssi-network/networks/dancebox/demo-evm-containerchain/){target=\_blank} deployed on Dancebox with an ID of `3001`. 
```

--bootnodes=/dns4/frag3-stagenet-relay-val-1.g.moondev.network/tcp/30334/p2p/12D3KooWQYLjopFtjojRBfTKkLFq2Untq9yG7gBjmAE8xcHFKbyq \
--bootnodes=/dns4/frag3-stagenet-relay-val-2.g.moondev.network/tcp/30334/p2p/12D3KooWMAtGe8cnVrg3qGmiwNjNaeVrpWaCTj82PGWN7PBx2tth \
--bootnodes=/dns4/frag3-stagenet-relay-val-3.g.moondev.network/tcp/30334/p2p/12D3KooWLKAf36uqBBug5W5KJhsSnn9JHFCcw8ykMkhQvW7Eus3U \
--bootnodes=/dns4/vira-stagenet-relay-validator-0.a.moondev.network/tcp/30334/p2p/12D3KooWSVTKUkkD4KBBAQ1QjAALeZdM3R2Kc2w5eFtVxbYZEGKd \
--bootnodes=/dns4/vira-stagenet-relay-validator-1.a.moondev.network/tcp/30334/p2p/12D3KooWFJoVyvLNpTV97SFqs91HaeoVqfFgRNYtUYJoYVbBweW4 \
--bootnodes=/dns4/vira-stagenet-relay-validator-2.a.moondev.network/tcp/30334/p2p/12D3KooWP1FA3dq1iBmEBYdQKAe4JNuzvEcgcebxBYMLKpTNirCR \
--bootnodes=/dns4/vira-stagenet-relay-validator-3.a.moondev.network/tcp/30334/p2p/12D3KooWDaTC6H6W1F4NkbaqK3Ema3jzc2BbhE2tyD3YEf84yNLE 

[Install]
WantedBy=multi-user.target
```

### Run Flags {: #run-flags }

The flags used in the `ExecStart` command can be adjusted according to your preferences and hardware configuration. The following ones are some of the most note-worthy:

> - `--name INSERT_NAME` - a human-readable name for this node

Reviewer: flags are mirrored between docker and systemd right?

Author: Oh, missed that!

Note that the `ExecStart` command  has some parameters that need to be changed to match your specific Appchain:

- `Specification file` - replace `YOUR_APPCHAIN_SPECS_FILE_LOCATION` with your Appchain's absolute path. If the file was copied in the same directory as the binary file and the relay chain specs, then your path will look like `/var/lib/appchain-data/YOUR_FILENAME.json`, e.g. `/var/lib/appchain-data/spec-raw.json`
- `Bootnode` - a bootnode is a full archive node that is used to sync the network from scratch. You'll need to [retrieve your Tanssi Appchain bootnode](#fetching-bootnode-information) and replace `INSERT_YOUR_APPCHAIN_BOOTNODE` with the actual bootnode information

=== "EVM-Compatible Appchain"

    ```bash
    [Unit]
    Description="Appchain systemd service"
    After=network.target
    StartLimitIntervalSec=0

    [Service]
    Type=simple
    Restart=on-failure
    RestartSec=10
    User=appchain_node_service
    SyslogIdentifier=appchain
    SyslogFacility=local7
    KillSignal=SIGHUP
    ExecStart=/var/lib/appchain-data/container-chain-template-frontier-node \

>     --chain=YOUR_APPCHAIN_SPECS_FILE_LOCATION \

Reviewer: This is still not matching the order here -> http://docs.tanssi.network/node-operators/appchain-node/rpc-docker/#start-up-command

Author: I changed the order also in the docker article (in this same PR) to match both commands
(The only remaining diff is that we are using "--base-path=/var/lib/appchain-data" in systemd to point to the directory created through the steps)
